### PR TITLE
announce and add notes for n30352

### PR DESCRIPTION
**Pull request by glozow**

### CODE REVIEW

File: _posts/2024-08-07-#30352.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  participants to CPFP the presigned transactions at broadcast time. The current design (see [BOLT
+  3](https://github.com/lightning/bolts/blob/master/03-transactions.md#to_local_anchor-and-to_remote_anchor-output-option_anchors))
+  has a few limitations described in [this blog post](https://bitcoinops.org/en/blog/waiting-for-confirmation/#policy-as-an-interface).
+
+    - The most relevant point here is the fact that the anchors currently specify a p2wsh script including two spending
+      paths: using the party's funding key, or anyone 16 blocks after the transaction confirms. Thus, using anchor
+      outputs requires knowing the funding key and paying fees for the relatively large amount of data that needs to be
+      included in the child transaction.
+
+    - Assuming there are not other reasons to use keyed anchor outputs (out of scope for this review club), a keyless
```
Comment (LarryRuane) [inline _posts/2024-08-07-#30352.md:25]:
```suggestion
    - Assuming there are no other reasons to use keyed anchor outputs (out of scope for this review club), a keyless
```

File: _posts/2024-08-07-#30352.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  presigned transactions, with a few key improvements:
+
+    - The anchor output can have any `nValue`, including amounts below the dust threshold such as 0, as long as it is
+      spent immediately by a fee-bumping child. Implementing this policy requires ensuring that a dusty anchor is always
+      spent after subsequent mempool updates, so it is only implemented for TRUC transactions. This portion of the
+      proposal was split into its own PR, [#30239](https://github.com/bitcoin/bitcoin/pull/30239).
+
+    - The anchor output is "keyless" or "anyone-can-spend", reducing the amount of data (and thus fees) needed in the
+      CPFP transaction, and making it easier for watchtowers to help broadcast presigned transactions. This part of the
+      proposal, [#30352](https://github.com/bitcoin/bitcoin/pull/30352), is independent from the "ephemeral dust"
```
Comment (LarryRuane) [inline _posts/2024-08-07-#30352.md:38]:
```suggestion
      proposal, [#30352](https://github.com/bitcoin/bitcoin/pull/30352), is independent of the "ephemeral dust"
```

File: _posts/2024-08-07-#30352.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+- This PR does two things: it defines `OP_1 <0x4e74>` as a new output type, and relaxes policy rules to make it standard
+  to spend this output type, as long as the witness is empty.
+
+- While `scriptPubKey`s can be fairly freeform, Bitcoin Core enumerates several `TxoutType`s. These correspond to output
+  types that you may be familiar with like `SCRIPTHASH` (P2SH), `WITNESS_V0_KEYHASH` (P2WPKH), `WITNESS_V0_SCRIPTHASH`
+  (P2WSH), and `NULL_DATA` (`OP_RETURN` or datacarrier).
+
+    - [`Solver`](https://github.com/bitcoin/bitcoin/blob/24f86783c87e836c98404bcc20a07742736d6b56/src/script/solver.cpp#L141) pattern-matches `scriptPubKey`s to classify their output type; anything that does not fall into the known categories is `TxoutType::NONSTANDARD`.
+
+    - By default, a transaction must pass standardness checks in order to be accepted to mempool. `IsStandardTx()`
```
Comment (LarryRuane) [inline _posts/2024-08-07-#30352.md:50]:
```suggestion
    - By default, a transaction must pass standardness checks to be accepted to mempool. `IsStandardTx()`
```

File: _posts/2024-08-07-#30352.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      inspects the `TxoutType`s of each of the transaction's [inputs and outputs](https://github.com/bitcoin/bitcoin/blob/fa0b5d68823b69f4861b002bbfac2fd36ed46356/src/policy/policy.cpp#L111-L148), among other checks.
+
+    - Notice the difference in rules applied to inputs and outputs. It's possible that a particular output type is
```
Comment (LarryRuane) [inline _posts/2024-08-07-#30352.md:53]:
```suggestion
    - Notice the difference in rules applied to inputs and outputs. A particular output type may be
```

File: _posts/2024-08-07-#30352.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      nonstandard to create but standard to spend, and vice versa.
+
+## Questions
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
+
+1. Before `TxoutType::ANCHOR` is defined in this PR, what `TxoutType` would a `scriptPubKey` `OP_1 <0x4e73>` be classified as? (Hint:
+   what would [`Solver`](https://github.com/bitcoin/bitcoin/blob/24f86783c87e836c98404bcc20a07742736d6b56/src/script/solver.cpp#L141) return?)
+
+1. Based on the answer to the previous question, would it be standard to create this output type? What about spend it?
```
Comment (LarryRuane) [inline _posts/2024-08-07-#30352.md:63]:
```suggestion
1. Based on the answer to the previous question, would it be standard to create this output type? What about to spend it?
```

File: _posts/2024-08-07-#30352.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+   (Hint: how do [`IsStandard`](https://github.com/bitcoin/bitcoin/blob/fa0b5d68823b69f4861b002bbfac2fd36ed46356/src/policy/policy.cpp#L70) and [`AreInputsStandard`](https://github.com/bitcoin/bitcoin/blob/fa0b5d68823b69f4861b002bbfac2fd36ed46356/src/policy/policy.cpp#L177) treat this type?)
+
+1. Before this PR, with default settings, what are the output types that can be _created_ in a standard transaction? Is
```
Comment (LarryRuane) [inline _posts/2024-08-07-#30352.md:66]:
```suggestion
1. Before this PR, with default settings, which output types that can be _created_ in a standard transaction? Is
```

File: _posts/2024-08-07-#30352.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+1. Why does the size of the output script of an anchor output matter?
+
+1. What other ways can you think of to implement an 'anyone-can-spend' anchor?
+
+1. What would be the problem with using `P2SH(OP_TRUE)`?
+
+1. What's the difference between `OP_TRUE` and `OP_1`? (Hint: where are they defined in the code?)
+
+1. What is the number of virtual bytes needed to create and spend this anchor output?
```
Comment (LarryRuane) [inline _posts/2024-08-07-#30352.md:81]:
```suggestion
1. How many virtual bytes are needed to create and spend this anchor output?
```

File: _posts/2024-08-07-#30352.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
+---
+layout: pr
+date: 2024-08-07
+title: "Add PayToAnchor(P2A), OP\_1 <0x4e73> as standard output sript for spending"
```
Comment (LarryRuane) [inline _posts/2024-08-07-#30352.md:4]:
```suggestion
title: "Add PayToAnchor(P2A), OP\_1 <0x4e73> as standard output script for spending"
```

**Review by glozow — COMMENTED** [review]
Thanks @LarryRuane :) accepted all your suggestions

**Review by stickies-v — COMMENTED** [review]
Very nice notes, thank you! I've pushed (hopefully) uncontroversial fixes as new commits and left the open ended ones as comments in this review.

File: _posts/2024-08-07-#30352.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+      output requires paying fees for a relatively large amount of data.
+
+    - Assuming there are no other reasons to use keyed anchor outputs (out of scope for this review club), a keyless
+      anchor may reduce complexity (including for a watchtower) and make fee-bumping more space and fee-efficient.
+
+- [Ephemeral Anchors](https://bitcoinops.org/en/topics/ephemeral-anchors/) enables a new pattern for adding fees to
+  presigned transactions, with a few key improvements:
+
+    - The anchor output can have any `nValue`, including amounts below the dust threshold such as 0, as long as it is
+      spent immediately by a fee-bumping child. Implementing this policy requires ensuring that a dusty anchor is always
```
[the remark this answers is not in the document]
Comment (glozow) [reply _posts/2024-08-07-#30352.md]: Added some of this, hopefully people who are confused read the link :shrug:

[the remark this answers is not in the document]
Comment (glozow) [reply _posts/2024-08-07-#30352.md]: just "the anchor" now

File: _posts/2024-08-07-#30352.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+      spent after subsequent mempool updates, so it is only implemented for TRUC transactions. This portion of the
+      proposal was split into its own PR, [#30239](https://github.com/bitcoin/bitcoin/pull/30239).
+
+    - The anchor output is "keyless" or "anyone-can-spend", reducing the amount of data (and thus fees) needed in the
+      CPFP transaction, and making it easier for watchtowers to help broadcast presigned transactions. This part of the
+      proposal, [#30352](https://github.com/bitcoin/bitcoin/pull/30352), is independent of the "ephemeral dust"
+      concept, and the implementation does not benefit from simpler topologies.
```
[the remark this answers is not in the document]
Comment (glozow) [reply _posts/2024-08-07-#30352.md]: made it "simple regardless of topology"

File: _posts/2024-08-07-#30352.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+1. Why does the size of the output script of an anchor output matter?
+
+1. What other ways can you think of to implement an 'anyone-can-spend' anchor?
+
+1. What would be the problem with using `P2SH(OP_TRUE)`?
+
+1. What is the difference between `OP_TRUE` and `OP_1`? (Hint: where are they defined in the code?)
+
+1. How many virtual bytes are needed to create and spend this anchor output?
```
[the remark this answers is not in the document]
Comment (glozow) [reply _posts/2024-08-07-#30352.md]: changed to p2a

**Comment by glozow** [comment]
Thanks @stickies-v! Your edits make sense to me. Addressed the comments in a new commit.

**Review by stickies-v — APPROVED** [review]
LGTM, thanks @glozow !

**Pull request merged**